server/routes.go

```diff
@@ -186,7 +175,26 @@ func (s *Server) GenerateHandler(c *gin.Context) {
 			b.WriteString(s)
 		}
 
-		if err := tmpl.Execute(&b, template.Values{Messages: msgs}); err != nil {
+		var values template.Values
+		if req.Suffix != "" {
```

Comment:
What's the logic to not applying system in this case? Feels like it could still potentially be useful if specified.

Reply:
This is a good catch – however are there any models that support this? I think most of the "context" goes into the prefix if I understand FIM prompting correctly

Reply:
most fim templates I've seen do not have a system prompt since they're geared towards completion tasks, mainly code completion. adding a system prompt, which is mostly a instruct/chat construct, breaks the flow of those templates